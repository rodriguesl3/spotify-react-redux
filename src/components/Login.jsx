import React from 'react'
import PropTypes from 'prop-types'
import './Login.scss'
import { withRouter } from 'react-router-dom'


const Login = (props) => {
    const { history } = props;

    const handleSubmit = () => {
        props.onSubmit(props, history)
    }

    return (
        <div id="container" >
            <img className="imgBackground" src="https://images.pexels.com/photos/167491/pexels-photo-167491.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260" alt="" />
            <div onClick={handleSubmit} className="pulse"></div>
            <div className="dot"></div>
        </div>
    );
}

Login.propTypes = {

}

export default withRouter(Login);