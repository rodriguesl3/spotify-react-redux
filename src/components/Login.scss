body {
    background-color: black;
    color: gray;
}

#container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100%;
    overflow: hidden;
}

.imgBackground {
    overflow: hidden;
    background-size: cover;
    background-position: center;
    width: 100%;
}



.pulse {
    width: 200px;
    height: 200px;
    border: 5px solid #28a745;
    border-radius: 100px;
    background-color: #1e7e34;
    z-index: 0;
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 25px;
    color: white;
    transition: .5s;
    cursor: pointer;
    z-index: 1;
}

.pulse::after {
    content: "Login";
}

.pulse:hover::after {
    content: "Vamos l√°?";
}

.pulse:hover {
    width: 300px;
    height: 100px;
}

.pulse:hover~.dot {
    width: 400px;
    height: 200px;
}



.dot {
    border: 10px solid black;
    background: transparent;
    border-radius: 650px;
    height: 400px;
    width: 400px;
    -webkit-animation: pulse 3s ease-out;
    animation: pulse 3s ease-out;
    -webkit-animation-iteration-count: infinite;
    animation-iteration-count: infinite;

    opacity: 0;
    position: absolute;
}







@-moz-keyframes pulse {
    0% {
        -moz-transform: scale(0);
        opacity: 0;
    }

    25% {
        -moz-transform: scale(0);
        opacity: 0.1;
    }

    50% {
        -moz-transform: scale(0.1);
        opacity: 0.3;
    }

    75% {
        -moz-transform: scale(0.5);
        opacity: 0.5;
    }

    100% {
        -moz-transform: scale(1);
        opacity: 0.1;
    }
}

@-webkit-keyframes "pulse" {
    0% {
        -webkit-transform: scale(0);
        opacity: 0;
    }

    25% {
        -webkit-transform: scale(0);
        opacity: 0.1;
    }

    50% {
        -webkit-transform: scale(0.1);
        opacity: 0.3;
    }

    75% {
        -webkit-transform: scale(0.5);
        opacity: 0.5;
    }

    100% {
        -webkit-transform: scale(1);
        opacity: 0.1;
    }
}