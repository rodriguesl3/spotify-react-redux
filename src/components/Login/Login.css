@charset "UTF-8";
#container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  height: 100vh;
  width: 100%;
  overflow: hidden;
}

.imgUp {
  width: 100%;
  background-size: cover;
}

.imgDown {
  width: 100%;
  background-size: cover;
}

.content {
  width: 100%;
  height: 100vh;
  position: absolute;
  -webkit-clip-path: polygon(0 0, 0% 100%, 100% 100%);
          clip-path: polygon(0 0, 0% 100%, 100% 100%);
  background-color: #161925;
  position: absolute;
}

.pulse {
  width: 200px;
  height: 200px;
  border: 5px solid #28a745;
  border-radius: 100px;
  background-color: #1e7e34;
  z-index: 0;
  position: absolute;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  font-size: 25px;
  color: white;
  -webkit-transition: .5s;
  transition: .5s;
  cursor: pointer;
  z-index: 1;
}

.pulse::after {
  content: "Login";
}

.pulse:hover::after {
  content: "Vamos l√°?";
}

.pulse:hover {
  width: 300px;
  height: 100px;
}

.pulse:hover ~ .dot {
  width: 400px;
  height: 200px;
}

.dot {
  border: 10px solid black;
  background: transparent;
  border-radius: 650px;
  height: 400px;
  width: 400px;
  -webkit-animation: pulse 3s ease-out;
  animation: pulse 3s ease-out;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  opacity: 0;
  position: absolute;
}

@-webkit-keyframes "pulse" {
  0% {
    -webkit-transform: scale(0);
    opacity: 0;
  }
  25% {
    -webkit-transform: scale(0);
    opacity: 0.1;
  }
  50% {
    -webkit-transform: scale(0.1);
    opacity: 0.3;
  }
  75% {
    -webkit-transform: scale(0.5);
    opacity: 0.5;
  }
  100% {
    -webkit-transform: scale(1);
    opacity: 0.1;
  }
}
/*# sourceMappingURL=Login.css.map */