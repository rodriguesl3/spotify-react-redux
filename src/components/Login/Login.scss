$primary-color: #161925;
$accent-color: #23395B;
$alternative-color: #406e8e;
$base-color:#8EA8C3;
$component-color:#CBF7ED;



#container {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    width: 100%;
    overflow: hidden;


}


.imgUp {
    width: 100%;
    background-size: cover;
}

.imgDown {
    width: 100%;
    background-size: cover;
}



.content {
    width: 100%;
    height: 100vh;
    position: absolute;
    clip-path: polygon(0 0, 0% 100%, 100% 100%);
    background-color: $primary-color;
    position: absolute;
}


.pulse {
    width: 200px;
    height: 200px;
    border: 5px solid #28a745;
    border-radius: 100px;
    background-color: #1e7e34;
    z-index: 0;
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 25px;
    color: white;
    transition: .5s;
    cursor: pointer;
    z-index: 1;
}

.pulse::after {
    content: "Login";
}

.pulse:hover::after {
    content: "Vamos l√°?";
}

.pulse:hover {
    width: 300px;
    height: 100px;
}

.pulse:hover~.dot {
    width: 400px;
    height: 200px;
}



.dot {
    border: 10px solid black;
    background: transparent;
    border-radius: 650px;
    height: 400px;
    width: 400px;
    -webkit-animation: pulse 3s ease-out;
    animation: pulse 3s ease-out;
    -webkit-animation-iteration-count: infinite;
    animation-iteration-count: infinite;

    opacity: 0;
    position: absolute;
}







@-moz-keyframes pulse {
    0% {
        -moz-transform: scale(0);
        opacity: 0;
    }

    25% {
        -moz-transform: scale(0);
        opacity: 0.1;
    }

    50% {
        -moz-transform: scale(0.1);
        opacity: 0.3;
    }

    75% {
        -moz-transform: scale(0.5);
        opacity: 0.5;
    }

    100% {
        -moz-transform: scale(1);
        opacity: 0.1;
    }
}

@-webkit-keyframes "pulse" {
    0% {
        -webkit-transform: scale(0);
        opacity: 0;
    }

    25% {
        -webkit-transform: scale(0);
        opacity: 0.1;
    }

    50% {
        -webkit-transform: scale(0.1);
        opacity: 0.3;
    }

    75% {
        -webkit-transform: scale(0.5);
        opacity: 0.5;
    }

    100% {
        -webkit-transform: scale(1);
        opacity: 0.1;
    }
}