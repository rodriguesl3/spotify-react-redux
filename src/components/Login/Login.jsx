import React from 'react';
import './Login.scss';
import { withRouter } from 'react-router-dom';


const Login = (props) => {
  const { history } = props;

  const handleSubmit = () => {
    props.onSubmit(props, history);
  };

  return (
    <div>
      <div className="content">
        <img src="https://wallpapercave.com/wp/EYS4oYf.jpg" className="imgUp" alt="" />
      </div>
      <div id="container">
        <img src="https://wallpapercave.com/wp/wp2130009.jpg" className="imgDown" alt="" />
        <div onClick={handleSubmit} className="pulse" />
        <div className="dot" />
      </div>
    </div>
  );
};

Login.propTypes = {

};

export default withRouter(Login);
