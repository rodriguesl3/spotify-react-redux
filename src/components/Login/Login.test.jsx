/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Login from './Login';

test('Login snaphot test', () => {
  const component = renderer.create(
    <MemoryRouter>
      <Login />
    </MemoryRouter>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
