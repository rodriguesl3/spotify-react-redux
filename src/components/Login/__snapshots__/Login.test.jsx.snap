// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snaphot test 1`] = `
<div>
  <div
    className="content"
  >
    <img
      alt=""
      className="imgUp"
      src="https://wallpapercave.com/wp/EYS4oYf.jpg"
    />
  </div>
  <div
    id="container"
  >
    <img
      alt=""
      className="imgDown"
      src="https://wallpapercave.com/wp/wp2130009.jpg"
    />
    <div
      className="pulse"
      onClick={[Function]}
      role="clickHandle"
    />
    <div
      className="dot"
    />
  </div>
</div>
`;
